app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java

```diff
@@ -918,28 +912,13 @@ private void handleGetDatum(CommCareSession session) {
     }
 
     private void handleCompute(CommCareSession session) {
-        SessionDatum datum = session.getNeededDatum();
-        XPathExpression form;
+        EvaluationContext ec = CommCareApplication._().getCurrentSessionWrapper().getEvaluationContext();
         try {
-            form = XPathParseTool.parseXPath(datum.getValue());
-        } catch (XPathSyntaxException e) {
-            //TODO: What.
-            e.printStackTrace();
-            throw new RuntimeException(e.getMessage());
-        }
-        EvaluationContext ec = session.getEvaluationContext(new CommCareInstanceInitializer(session));
-        if (datum.getType() == SessionDatum.DATUM_TYPE_FORM) {
-            session.setXmlns(XPathFuncExpr.toString(form.eval(ec)));
-            session.setDatum("", "awful");
-        } else {
-            try {
-                session.setDatum(datum.getDataId(), XPathFuncExpr.toString(form.eval(ec)));
-            } catch (XPathException e) {
-                displayException(e);
-                return;
-            }
+            session.setComputedDatum(ec);
+            startNextFetch();
+        } catch (XPathException e) {
+            displayException(e);
```

Comment:
No-- when the dialog opened by displayException() is closed, it just calls finish()